packages/kbn-expandable-flyout/src/reducer.ts

```diff
@@ -6,107 +6,62 @@
  * Side Public License, v 1.
  */
 
+import { createReducer } from '@reduxjs/toolkit';
 import { FlyoutPanelProps } from './types';
-import { Action, ActionType } from './actions';
+import {
+  openPanelsAction,
+  openLeftPanelAction,
+  openRightPanelAction,
+  closePanelsAction,
+  closeLeftPanelAction,
+  closePreviewPanelAction,
+  closeRightPanelAction,
+  previousPreviewPanelAction,
+  openPreviewPanelAction,
+} from './actions';
+import { initialState } from './state';
 
-export interface State {
-  /**
-   * Panel to render in the left section
-   */
-  left: FlyoutPanelProps | undefined;
-  /**
-   * Panel to render in the right section
-   */
-  right: FlyoutPanelProps | undefined;
-  /**
-   * Panels to render in the preview section
-   */
-  preview: FlyoutPanelProps[];
-}
+export const reducer = createReducer(initialState, (builder) => {
+  builder.addCase(openPanelsAction, (state, { payload: { preview, left, right } }) => {
+    state.preview = preview ? [preview] : [];
+    state.right = right;
+    state.left = left;
+  });
 
-export const initialState: State = {
-  left: undefined,
-  right: undefined,
-  preview: [],
-};
+  builder.addCase(openLeftPanelAction, (state, { payload }) => {
+    state.left = payload;
+  });
 
-export function reducer(state: State, action: Action) {
-  switch (action.type) {
-    /**
-     * Open the flyout by replacing the entire state with new panels.
-     */
-    case ActionType.openFlyout: {
-      const { left, right, preview } = action.payload;
-      return {
-        left,
-        right,
-        preview: preview ? [preview] : [],
-      };
-    }
+  builder.addCase(openRightPanelAction, (state, { payload }) => {
+    state.right = payload;
+  });
 
-    /**
-     * Opens a right section by replacing the previous right panel with the new one.
-     */
-    case ActionType.openRightPanel: {
-      return { ...state, right: action.payload };
-    }
+  builder.addCase(openPreviewPanelAction, (state, { payload }) => {
+    state.preview = [...state.preview, payload];
```

Comment:
Thanks to [Immer,](https://redux-toolkit.js.org/usage/immer-reducers#redux-toolkit-and-immer) we can simplify this to:

```js
state.preview.push(payload)
```

Reply:
done, thanks